### PR TITLE
Add proxy support to build.sh

### PR DESCRIPTION
**Pull request by mchlumsky**
Add the ability to specify a proxy via environment variables for http and https requests done during the docker build process.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -113,6 +117,12 @@ DOCKER_BUILD_ARGS=(
   --build-arg "DOCKER_ORG=${DOCKER_ORG}"
   --build-arg "DOCKER_REPO=${DOCKER_REPO}"
 )
+if [ -n "HTTP_PROXY" ]; then
+  DOCKER_BUILD_ARGS+=( --build-arg "http_proxy=${HTTP_PROXY}" )
```
Comment (cimnine) [inline build.sh]: For the sake of consistence, could you try to use the upper-case equivalent of `http_proxy`, i.e. `HTTP_PROXY`? This should be no problem per my interpretation of the [Predefined Build Args](https://docs.docker.com/engine/reference/builder/#predefined-args) documentation.

File: build.sh [hunk 2]
```diff
@@ -49,6 +49,10 @@ if [ "${1}x" == "x" ] || [ "${1}" == "--help" ] || [ "${1}" == "-h" ]; then
   echo "           Example: VARIANT=ldap will result in the tag 'latest-ldap' and the"
   echo "            Dockerfile 'Dockerfile.ldap' being used."
   echo "           Default: empty"
+  echo "  HTTP_PROXY The proxy to use for http requests."
+  echo "           Default: empty"
+  echo "  HTTPS_PROXY The proxy to use for https requests."
+  echo "           Default: empty"
```
Comment (cimnine) [inline build.sh]: If you're already at it, could you add the `FTP_PROXY` as well?

**Comment by mchlumsky** [comment]
Should I also add NO_PROXY while I'm at it?

**Comment by cimnine** [comment]
I did not see the purpose of NO_PROXY. What value would it serve?

**Comment by mchlumsky** [comment]
Generally, NO_PROXY is useful when you have some restrictive proxy that only allows you to use it for external traffic but blocks access to internal resources. For example, you would set something like NO_PROXY=192.168.1.* to prevent your application from using the proxy to access anything with an IP in the 192.168.1.0/24 range.

The reason for this PR was because pip couldn't fetch packages from pypi. We also have an internal pypi where the proxy must not be used, so it could be a use-case for NO_PROXY (although it's not a use-case right now).

File: build.sh (hunk 1 above)
Comment (mchlumsky) [reply build.sh]: So it turns out only the lowercase forms of these environment variables are taken into account by both wget and pip.
I kept the variable names uppercase when calling build.sh to match the other variables being read by build.sh however I set the lowercase ones with --build-arg so they get properly used inside the docker build.

**Comment by cimnine** [comment]
Thanks for explaining NO_PROXY. I guess it would make sense to add it then. Could you also rebase the branch on the current master, or merge the current master into your branch?

**Comment by mchlumsky** [comment]
Rebased and squashed into 1 commit.
Thanks for accepting the PR!

File: build.sh [hunk 3]
```diff
@@ -49,6 +49,14 @@ if [ "${1}x" == "x" ] || [ "${1}" == "--help" ] || [ "${1}" == "-h" ]; then
...
```
Comment (cimnine) [inline build.sh]: One more thing: Could you maybe add an example to at least one of these?

**Comment by cimnine** [comment]
Thanks again for your PR!

**Pull request merged**